### ISSUE TITLE
When Xcode 10 parallel testing is enabled, test failures look like build failures

### ISSUE THREAD

**Issue by benasher44** [fastlane/fastlane#13414]
🔑

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When using Xcode 10 parallel testing and a test fails, fastlane exits with the following message:

`Error building/testing the application - see the log above`

This is unexpected because we have `fail_build` set to false, and this lane is exiting due to a test failure.

Here's the output from Xcode a few lines up in our CI log:

```
Failing tests:
IssuesLogUITests.testSingleIssueChanges()
** TEST EXECUTE FAILED **
[17:42:02]: Exit status: 65
```

##### Complete output when running fastlane, including the stack trace and command used
```
run_tests(
      test_without_building: true,
      only_testing: ['PlanGridUITests'],
      code_coverage: true,
      output_types: '',
      fail_build: false
)
```
<details>
  <pre>[plangrid] Running shell script
+ bundle exec fastlane uitest
+-------------------------+---------+---------------------------+
|                         Used plugins                          |
+-------------------------+---------+---------------------------+
| Plugin                  | Version | Action                    |
+-------------------------+---------+---------------------------+
| fastlane-plugin-trainer | 0.4.0   | trainer                   |
| fastlane-plugin-bugsnag | 1.3.2   | send_build_to_bugsnag     |
|                         |         | upload_symbols_to_bugsnag |
+-------------------------+---------+---------------------------+

[17:18:42]: ------------------------------
[17:18:42]: --- Step: default_platform ---
[17:18:42]: ------------------------------
[17:18:42]: Driving the lane 'ios uitest' 🚀
[17:18:42]: ----------------------
[17:18:42]: --- Step: hostname ---
[17:18:42]: ----------------------
[17:18:42]: $ hostname
[17:18:42]: ▸ redacted
[17:18:42]: --------------------------------------------
[17:18:42]: --- Step: ipconfig getifaddr en0 || true ---
[17:18:42]: --------------------------------------------
[17:18:42]: $ ipconfig getifaddr en0 || true
[17:18:42]: ▸ redacted
[17:18:42]: --------------------------------------
[17:18:42]: --- Step: xcodebuild -version -sdk ---
[17:18:42]: --------------------------------------
[17:18:42]: $ xcodebuild -version -sdk
[17:18:42]: ▸ iPhoneOS12.0.sdk - iOS 12.0 (iphoneos12.0)
[17:18:42]: ▸ SDKVersion: 12.0
[17:18:42]: ▸ Path: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.0.sdk
[17:18:42]: ▸ PlatformVersion: 12.0
[17:18:42]: ▸ PlatformPath: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform
[17:18:42]: ▸ BuildID: A8482458-AB48-11E8-BDA0-0514931EADDD
[17:18:42]: ▸ ProductBuildVersion: 16A366
[17:18:42]: ▸ ProductCopyright: 1983-2018 Apple Inc.
[17:18:42]: ▸ ProductName: iPhone OS
[17:18:42]: ▸ ProductVersion: 12.0
[17:18:42]: ▸ iPhoneSimulator12.0.sdk - Simulator - iOS 12.0 (iphonesimulator12.0)
[17:18:42]: ▸ SDKVersion: 12.0
[17:18:42]: ▸ Path: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.0.sdk
[17:18:42]: ▸ PlatformVersion: 12.0
[17:18:42]: ▸ PlatformPath: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform
[17:18:42]: ▸ BuildID: A8482458-AB48-11E8-BDA0-0514931EADDD
[17:18:42]: ▸ ProductBuildVersion: 16A366
[17:18:42]: ▸ ProductCopyright: 1983-2018 Apple Inc.
[17:18:42]: ▸ ProductName: iPhone OS
[17:18:42]: ▸ ProductVersion: 12.0
[17:18:42]: ▸ MacOSX10.14.sdk - macOS 10.14 (macosx10.14)
[17:18:42]: ▸ SDKVersion: 10.14
[17:18:42]: ▸ Path: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
[17:18:42]: ▸ PlatformVersion: 1.1
[17:18:42]: ▸ PlatformPath: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform
[17:18:42]: ▸ ProductBuildVersion: 18A384
[17:18:42]: ▸ ProductCopyright: 1983-2018 Apple Inc.
[17:18:42]: ▸ ProductName: Mac OS X
[17:18:42]: ▸ ProductUserVisibleVersion: 10.14
[17:18:42]: ▸ ProductVersion: 10.14
[17:18:42]: ▸ AppleTVOS12.0.sdk - tvOS 12.0 (appletvos12.0)
[17:18:42]: ▸ SDKVersion: 12.0
[17:18:42]: ▸ Path: /Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS12.0.sdk
[17:18:42]: ▸ PlatformVersion: 12.0
[17:18:42]: ▸ PlatformPath: /Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform
[17:18:42]: ▸ BuildID: 8CB266D2-AB47-11E8-AD6C-B115D1DC00F8
[17:18:42]: ▸ ProductBuildVersion: 16J364
[17:18:42]: ▸ ProductCopyright: 1983-2018 Apple Inc.
[17:18:42]: ▸ ProductName: Apple TVOS
[17:18:42]: ▸ ProductVersion: 12.0
[17:18:42]: ▸ AppleTVSimulator12.0.sdk - Simulator - tvOS 12.0 (appletvsimulator12.0)
[17:18:42]: ▸ SDKVersion: 12.0
[17:18:42]: ▸ Path: /Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator12.0.sdk
[17:18:42]: ▸ PlatformVersion: 12.0
[17:18:42]: ▸ PlatformPath: /Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform
[17:18:42]: ▸ BuildID: 8CB266D2-AB47-11E8-AD6C-B115D1DC00F8
[17:18:42]: ▸ ProductBuildVersion: 16J364
[17:18:42]: ▸ ProductCopyright: 1983-2018 Apple Inc.
[17:18:42]: ▸ ProductName: Apple TVOS
[17:18:42]: ▸ ProductVersion: 12.0
[17:18:42]: ▸ WatchOS5.0.sdk - watchOS 5.0 (watchos5.0)
[17:18:42]: ▸ SDKVersion: 5.0
[17:18:42]: ▸ Path: /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS5.0.sdk
[17:18:42]: ▸ PlatformVersion: 5.0
[17:18:42]: ▸ PlatformPath: /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform
[17:18:42]: ▸ BuildID: A011E0F6-AB4A-11E8-A58A-04145DEEDE81
[17:18:42]: ▸ ProductBuildVersion: 16R363
[17:18:42]: ▸ ProductCopyright: 1983-2018 Apple Inc.
[17:18:42]: ▸ ProductName: Watch OS
[17:18:42]: ▸ ProductVersion: 5.0
[17:18:42]: ▸ WatchSimulator5.0.sdk - Simulator - watchOS 5.0 (watchsimulator5.0)
[17:18:42]: ▸ SDKVersion: 5.0
[17:18:42]: ▸ Path: /Applications/Xcode.app/Contents/Developer/Platforms/WatchSimulator.platform/Developer/SDKs/WatchSimulator5.0.sdk
[17:18:42]: ▸ PlatformVersion: 5.0
[17:18:42]: ▸ PlatformPath: /Applications/Xcode.app/Contents/Developer/Platforms/WatchSimulator.platform
[17:18:42]: ▸ BuildID: A011E0F6-AB4A-11E8-A58A-04145DEEDE81
[17:18:42]: ▸ ProductBuildVersion: 16R363
[17:18:42]: ▸ ProductCopyright: 1983-2018 Apple Inc.
[17:18:42]: ▸ ProductName: Watch OS
[17:18:42]: ▸ ProductVersion: 5.0
[17:18:42]: ▸ Xcode 10.0
[17:18:42]: ▸ Build version 10A255
[17:18:42]: -----------------------------
[17:18:42]: --- Step: unlock_keychain ---
[17:18:42]: -----------------------------
[17:18:42]: Unlocking keychain at path: /Users/jenkins/Library/Keychains/login.keychain-db
[17:18:42]: --------------------------------------
[17:18:42]: --- Step: Switch to ios reset lane ---
[17:18:42]: --------------------------------------
[17:18:42]: Cruising over to lane 'ios reset' 🚖
[17:18:42]: ---------------------------------------
[17:18:42]: --- Step: killall Simulator || true ---
[17:18:42]: ---------------------------------------
[17:18:42]: $ killall Simulator || true
[17:18:42]: ▸ No matching processes belonging to you were found
[17:18:43]: Resetting iPhone 5
[17:18:43]: Resetting iPhone 5s
[17:18:43]: Resetting iPhone 6
[17:18:44]: Resetting iPhone 6 Plus
[17:18:44]: Resetting iPhone 6s
[17:18:44]: Resetting iPhone 6s Plus
[17:18:45]: Resetting iPhone 7
[17:18:45]: Resetting iPhone 7 Plus
[17:18:45]: Resetting iPhone SE
[17:18:46]: Resetting iPad Air
[17:18:46]: Resetting iPad Air 2
[17:18:47]: Resetting iPad (5th generation)
[17:18:47]: Resetting iPad Pro (9.7 inch)
[17:18:47]: Resetting iPad Pro (12.9 inch)
[17:18:48]: Resetting iPad Pro (12.9-inch) (2nd generation)
[17:18:48]: Resetting iPad Pro (10.5-inch)
[17:18:48]: Resetting iPhone 5s
[17:18:49]: Resetting iPhone 6
[17:18:49]: Resetting iPhone 6 Plus
[17:18:49]: Resetting iPhone 6s
[17:18:49]: Resetting iPhone 6s Plus
[17:18:50]: Resetting iPhone 7
[17:18:50]: Resetting iPhone 7 Plus
[17:18:50]: Resetting iPhone 8
[17:18:51]: Resetting iPhone 8 Plus
[17:18:51]: Resetting iPhone SE
[17:18:51]: Resetting iPhone X
[17:18:52]: Resetting iPhone XR
[17:18:52]: Resetting iPhone XS
[17:18:52]: Resetting iPhone XS Max
[17:18:53]: Resetting iPad Air
[17:18:53]: Resetting iPad Air 2
[17:18:53]: Resetting iPad (5th generation)
[17:18:54]: Resetting iPad Pro (9.7-inch)
[17:18:54]: Resetting iPad Pro (12.9-inch)
[17:18:54]: Resetting iPad Pro (12.9-inch) (2nd generation)
[17:18:55]: Resetting iPad Pro (10.5-inch)
[17:18:55]: Resetting iPad (6th generation)
[17:18:56]: Cruising back to lane 'ios uitest' 🚘
[17:18:56]: -----------------------
[17:18:56]: --- Step: run_tests ---
[17:18:56]: -----------------------
[17:18:56]: Successfully loaded '/private/var/lib/jenkins/plangrid/fastlane/Scanfile' 📄

+-------------------+-------------------------+
| Detected Values from './fastlane/Scanfile'  |
+-------------------+-------------------------+
| workspace         | PlanGrid.xcworkspace    |
| configuration     | Debug                   |
| scheme            | PlanGrid_CI             |
| devices           | ["iPad Air 2"]          |
| derived_data_path | DerivedData             |
| formatter         | xcpretty-json-formatter |
| skip_slack        | true                    |
+-------------------+-------------------------+

[17:18:56]: $ xcodebuild -showBuildSettings -workspace PlanGrid.xcworkspace -scheme PlanGrid_CI -configuration Debug
[17:18:59]: Command timed out after 3 seconds on try 1 of 4, trying again with a 6 second timeout...
[17:19:03]: Successfully loaded '/private/var/lib/jenkins/plangrid/fastlane/Scanfile' 📄

+-------------------+-------------------------+
| Detected Values from './fastlane/Scanfile'  |
+-------------------+-------------------------+
| workspace         | PlanGrid.xcworkspace    |
| configuration     | Debug                   |
| scheme            | PlanGrid_CI             |
| devices           | ["iPad Air 2"]          |
| derived_data_path | DerivedData             |
| formatter         | xcpretty-json-formatter |
| skip_slack        | true                    |
+-------------------+-------------------------+

[17:19:03]: $ xcodebuild -showBuildSettings -workspace PlanGrid.xcworkspace -scheme PlanGrid_CI -configuration Debug
[17:19:06]: Command timed out after 3 seconds on try 1 of 4, trying again with a 6 second timeout...

+----------------------------+-------------------------+
|               Summary for scan 2.104.0               |
+----------------------------+-------------------------+
| test_without_building      | true                    |
| only_testing               | ["PlanGridUITests"]     |
| code_coverage              | true                    |
| fail_build                 | false                   |
| workspace                  | PlanGrid.xcworkspace    |
| devices                    | ["iPad Air 2"]          |
| skip_detect_devices        | false                   |
| scheme                     | PlanGrid_CI             |
| clean                      | false                   |
| skip_build                 | false                   |
| output_directory           | ./fastlane/test_output  |
| buildlog_path              | ~/Library/Logs/scan     |
| include_simulator_logs     | false                   |
| formatter                  | xcpretty-json-formatter |
| disable_concurrent_testing | false                   |
| derived_data_path          | DerivedData             |
| should_zip_build_products  | false                   |
| result_bundle              | false                   |
| open_report                | false                   |
| configuration              | Debug                   |
| skip_slack                 | true                    |
| slack_only_on_failure      | false                   |
| use_clang_report_name      | false                   |
| xcode_path                 | /Applications/Xcode.app |
+----------------------------+-------------------------+

[17:19:10]: $ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -workspace PlanGrid.xcworkspace -scheme PlanGrid_CI -configuration Debug -destination 'platform=iOS Simulator,id=67B15F95-D9D5-4228-AB74-19EA0DE255B4' -derivedDataPath 'DerivedData' -enableCodeCoverage YES -only-testing:PlanGridUITests test-without-building | tee '/Users/jenkins/Library/Logs/scan/PlanGrid-PlanGrid_CI.log' | xcpretty -f `xcpretty-json-formatter` --report junit --output '/var/folders/xr/b47lgdqj5nj1xy2g_jtpdrc40000gn/T/junit_report20180920-74106-1uiuvao'
[17:19:10]: ▸ Loading...
[17:19:13]: ▸ 2018-09-20 17:19:13.196 xcodebuild[74267:377521]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
[17:19:13]: ▸ /var/lib/jenkins/plangrid/DerivedData/Logs/Test/Test-PlanGrid_CI-2018.09.20_17-19-12--0700.xcresult/1_Test/Diagnostics/PlanGridUITests-F5D094AC-60C0-44B7-B530-4EB18FC2DDCA/PlanGridUITests-A24943D8-6FF2-44F9-85F0-9AF12AAB0286/Session-PlanGridUITests-2018-09-20_171913-8389mm.log
[17:19:13]: ▸ 2018-09-20 17:19:13.196 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: (6ED2248C-E3B5-4437-BB64-1B4FFEDDCB6A) Beginning test session PlanGridUITests-6ED2248C-E3B5-4437-BB64-1B4FFEDDCB6A at 2018-09-20 17:19:13.197 with Xcode 10A255 on target <DVTiPhoneSimulator: 0x7fc4b21074f0> {
[17:19:13]: ▸ SimDevice: Clone 1 of iPad Air 2 (03C3A343-3843-46D4-BC99-C6704D1F608B, iOS 12.0, Shutdown)
[17:19:13]: ▸ } (12.0 (16A366))
[17:19:33]: ▸ 2018-09-20 17:19:33.522 xcodebuild[74267:377532]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
[17:19:33]: ▸ /var/lib/jenkins/plangrid/DerivedData/Logs/Test/Test-PlanGrid_CI-2018.09.20_17-19-12--0700.xcresult/1_Test/Diagnostics/PlanGridUITests-F5D094AC-60C0-44B7-B530-4EB18FC2DDCA/PlanGridUITests-8E47EDDD-B4D4-48BD-9BC2-99792F072F8A/Session-PlanGridUITests-2018-09-20_171933-6aiORu.log
[17:19:33]: ▸ 2018-09-20 17:19:33.524 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: (F59A0137-94FF-4317-8322-CDA9E3BBC366) Beginning test session PlanGridUITests-F59A0137-94FF-4317-8322-CDA9E3BBC366 at 2018-09-20 17:19:33.524 with Xcode 10A255 on target <DVTiPhoneSimulator: 0x7fc4b29b5350> {
[17:19:33]: ▸ SimDevice: Clone 2 of iPad Air 2 (B320FE62-821C-44BD-AFEA-3DE055BE7C43, iOS 12.0, Shutdown)
[17:19:33]: ▸ } (12.0 (16A366))
[17:41:45]: ▸ 2018-09-20 17:41:45.332 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: 1352.393 elapsed -- Testing started completed.
[17:41:45]: ▸ 2018-09-20 17:41:45.332 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
[17:41:45]: ▸ 2018-09-20 17:41:45.332 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: 1352.393 sec, +1352.393 sec -- end
[17:41:45]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/0A4A09E7-4B07-4977-AD50-F3CB8ABC3CD6-75159.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:45]: ▸ Make sure that all profile data to be merged is generated from the same binary.
[17:41:45]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/00ED57E5-A739-4CB2-9EA7-A2EB6D907B61-74603.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:45]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/27A6C9B8-354C-475B-92FD-8FC299D561D0-75403.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:45]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/0486B104-24B4-43C6-B3DB-A007E6DC3AC0-75526.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:45]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/2B937781-05B5-4C9B-82DA-51FF112044D5-75611.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:45]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/3E7A4143-C917-46E8-8DDD-6BC74F450810-75281.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/3D0BC7E3-5D4E-4E67-93C4-EE01CB63126D-74701.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/5A9EF47C-E289-4DB8-BF8B-57A1975B059A-74468.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/5EA1705E-0BAE-4839-96DD-8834F2B871E1-74826.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/6714F729-467F-4263-9D59-7EB23B2CAFE9-74932.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/72FF9C0A-2B25-484E-9B80-5516ECCBB976-74548.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/72EAB102-DD73-482D-85B5-FB244D6A2428-75491.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/95C8E434-4F09-489C-8CE5-58424F088196-75461.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/A29D6B43-CFFE-4F2F-81D1-4128927D803F-75222.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/A70319FD-AB31-458B-9AE0-EEDC9871AE72-75293.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/A9F5425B-070B-449A-99E6-9B10927C3032-74791.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/AC0BBBB3-1BA1-4EE0-A21E-E65F0D50B042-75062.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/ADD5E16E-A5E8-4B39-B182-57516D90FF5E-74989.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:46]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/B3B00B41-A64D-424C-8EC4-F982F26BB54F-75340.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:47]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/B4C3A8F6-4B10-4C42-801F-C99AE3155B60-74705.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:47]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/BED80C46-5CAE-4622-A19A-F500C7A98099-75080.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:47]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/C754C107-ADF6-45F4-B98D-18AE5170E7E3-74764.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:47]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/DC7AF0B9-550F-4EE1-817D-32AC9BE3E59A-74954.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:47]: ▸ /var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/E831EE33-425E-4F0D-9269-1713F8DDE2AD-75677.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
[17:41:50]: ▸ Failing tests:
[17:41:50]: ▸ IssuesLogUITests.testSingleIssueChanges()
[17:41:50]: ▸ ** TEST EXECUTE FAILED **
2018-09-20 17:19:13.196 xcodebuild[74267:377521]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/var/lib/jenkins/plangrid/DerivedData/Logs/Test/Test-PlanGrid_CI-2018.09.20_17-19-12--0700.xcresult/1_Test/Diagnostics/PlanGridUITests-F5D094AC-60C0-44B7-B530-4EB18FC2DDCA/PlanGridUITests-A24943D8-6FF2-44F9-85F0-9AF12AAB0286/Session-PlanGridUITests-2018-09-20_171913-8389mm.log
2018-09-20 17:19:13.196 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: (6ED2248C-E3B5-4437-BB64-1B4FFEDDCB6A) Beginning test session PlanGridUITests-6ED2248C-E3B5-4437-BB64-1B4FFEDDCB6A at 2018-09-20 17:19:13.197 with Xcode 10A255 on target <DVTiPhoneSimulator: 0x7fc4b21074f0> {
SimDevice: Clone 1 of iPad Air 2 (03C3A343-3843-46D4-BC99-C6704D1F608B, iOS 12.0, Shutdown)
} (12.0 (16A366))
2018-09-20 17:19:33.522 xcodebuild[74267:377532]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/var/lib/jenkins/plangrid/DerivedData/Logs/Test/Test-PlanGrid_CI-2018.09.20_17-19-12--0700.xcresult/1_Test/Diagnostics/PlanGridUITests-F5D094AC-60C0-44B7-B530-4EB18FC2DDCA/PlanGridUITests-8E47EDDD-B4D4-48BD-9BC2-99792F072F8A/Session-PlanGridUITests-2018-09-20_171933-6aiORu.log
2018-09-20 17:19:33.524 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: (F59A0137-94FF-4317-8322-CDA9E3BBC366) Beginning test session PlanGridUITests-F59A0137-94FF-4317-8322-CDA9E3BBC366 at 2018-09-20 17:19:33.524 with Xcode 10A255 on target <DVTiPhoneSimulator: 0x7fc4b29b5350> {
SimDevice: Clone 2 of iPad Air 2 (B320FE62-821C-44BD-AFEA-3DE055BE7C43, iOS 12.0, Shutdown)
} (12.0 (16A366))
2018-09-20 17:41:45.332 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: 1352.393 elapsed -- Testing started completed.
2018-09-20 17:41:45.332 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2018-09-20 17:41:45.332 xcodebuild[74267:377473] [MT] IDETestOperationsObserverDebug: 1352.393 sec, +1352.393 sec -- end
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/0A4A09E7-4B07-4977-AD50-F3CB8ABC3CD6-75159.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
Make sure that all profile data to be merged is generated from the same binary.
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/00ED57E5-A739-4CB2-9EA7-A2EB6D907B61-74603.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/27A6C9B8-354C-475B-92FD-8FC299D561D0-75403.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/0486B104-24B4-43C6-B3DB-A007E6DC3AC0-75526.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/2B937781-05B5-4C9B-82DA-51FF112044D5-75611.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/3E7A4143-C917-46E8-8DDD-6BC74F450810-75281.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/3D0BC7E3-5D4E-4E67-93C4-EE01CB63126D-74701.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/5A9EF47C-E289-4DB8-BF8B-57A1975B059A-74468.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/5EA1705E-0BAE-4839-96DD-8834F2B871E1-74826.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/6714F729-467F-4263-9D59-7EB23B2CAFE9-74932.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/72FF9C0A-2B25-484E-9B80-5516ECCBB976-74548.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/72EAB102-DD73-482D-85B5-FB244D6A2428-75491.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/95C8E434-4F09-489C-8CE5-58424F088196-75461.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/A29D6B43-CFFE-4F2F-81D1-4128927D803F-75222.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/A70319FD-AB31-458B-9AE0-EEDC9871AE72-75293.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/A9F5425B-070B-449A-99E6-9B10927C3032-74791.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/AC0BBBB3-1BA1-4EE0-A21E-E65F0D50B042-75062.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/ADD5E16E-A5E8-4B39-B182-57516D90FF5E-74989.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/B3B00B41-A64D-424C-8EC4-F982F26BB54F-75340.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/B4C3A8F6-4B10-4C42-801F-C99AE3155B60-74705.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/BED80C46-5CAE-4622-A19A-F500C7A98099-75080.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/C754C107-ADF6-45F4-B98D-18AE5170E7E3-74764.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/DC7AF0B9-550F-4EE1-817D-32AC9BE3E59A-74954.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
/var/lib/jenkins/plangrid/DerivedData/Build/ProfileData/67B15F95-D9D5-4228-AB74-19EA0DE255B4/E831EE33-425E-4F0D-9269-1713F8DDE2AD-75677.profraw: <unknown>:__ntd_UITransitionContextViewControllerKey_<invalid loc>: Function basic block count change detected (counter mismatch)
Failing tests:
IssuesLogUITests.testSingleIssueChanges()
** TEST EXECUTE FAILED **
[17:42:02]: Exit status: 65
+----------------------------+---------------------------------------------------------------------------------------------------+
|                                                          Lane Context                                                          |
+----------------------------+---------------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM           | ios                                                                                               |
| PLATFORM_NAME              | ios                                                                                               |
| LANE_NAME                  | ios uitest                                                                                        |
| SCAN_DERIVED_DATA_PATH     | DerivedData                                                                                       |
| SCAN_GENERATED_PLIST_FILES | ["DerivedData/Logs/Test/Test-PlanGrid_CI-2018.09.20_17-19-12--0700.xcresult/TestSummaries.plist"] |
| SCAN_GENERATED_PLIST_FILE  | DerivedData/Logs/Test/Test-PlanGrid_CI-2018.09.20_17-19-12--0700.xcresult/TestSummaries.plist     |
+----------------------------+---------------------------------------------------------------------------------------------------+
[17:42:02]: Error building/testing the application - see the log above

+------+--------------------------------+-------------+
|                  fastlane summary                   |
+------+--------------------------------+-------------+
| Step | Action                         | Time (in s) |
+------+--------------------------------+-------------+
| 1    | default_platform               | 0           |
| 2    | hostname                       | 0           |
| 3    | ipconfig getifaddr en0 || true | 0           |
| 4    | xcodebuild -version -sdk       | 0           |
| 5    | unlock_keychain                | 0           |
| 6    | Switch to ios reset lane       | 0           |
| 7    | killall Simulator || true      | 0           |
| 💥   | run_tests                      | 1386        |
+------+--------------------------------+-------------+

+---------+--------------+----------------+
|        Plugin updates available         |
+---------+--------------+----------------+
| Plugin  | Your Version | Latest Version |
+---------+--------------+----------------+
| bugsnag | 1.3.2        | 1.3.4          |
+---------+--------------+----------------+
[17:42:02]: To update all plugins, just run
[17:42:02]: $ bundle exec fastlane update_plugins

[17:42:02]: fastlane finished with errors

[!] Error building/testing the application - see the log above</pre>
</details>

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                 |
| --------------------------- | ----------------------------------------------------- |
| OS                          | 10.13.6                                               |
| Ruby                        | 2.4.0                                                 |
| Bundler?                    | true                                                  |
| Git                         | git version 2.17.1 (Apple Git-112)                    |
| Installation Source         | ~/Code/plangrid/vendor/bundle/ruby/2.4.0/bin/fastlane |
| Host                        | Mac OS X 10.13.6 (17G65)                              |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.0/lib                          |
| OpenSSL Version             | OpenSSL 1.1.0e  16 Feb 2017                           |
| Is contained                | false                                                 |
| Is homebrew                 | false                                                 |
| Is installed via Fabric.app | false                                                 |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/           |
| Xcode Version               | 10.0                                                  |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.105.1 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status      |
| ----------------------- | ------- | ------------------ |
| fastlane-plugin-trainer | 0.4.0   | ✅ Up-To-Date       |
| fastlane-plugin-bugsnag | 1.3.2   | 🚫 Update available |


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.16.3       |
| io-console                | 0.4.6        |
| rake                      | 12.3.1       |
| CFPropertyList            | 3.0.0        |
| concurrent-ruby           | 1.0.5        |
| i18n                      | 0.9.5        |
| minitest                  | 5.11.3       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.5        |
| activesupport             | 4.2.10       |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| artifactory               | 2.8.2        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| cork                      | 0.3.0        |
| nap                       | 1.1.0        |
| open4                     | 1.3.4        |
| claide-plugins            | 0.9.2        |
| fuzzy_match               | 2.0.4        |
| cocoapods-core            | 1.5.3        |
| cocoapods-deintegrate     | 1.0.2        |
| cocoapods-downloader      | 1.2.1        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.0        |
| cocoapods-try             | 1.1.0        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.1.3        |
| molinillo                 | 0.6.5        |
| ruby-macho                | 1.1.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.6.0        |
| cocoapods                 | 1.5.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.3       |
| faraday-http-cache        | 1.3.1        |
| git                       | 1.4.0        |
| kramdown                  | 1.16.2       |
| no_proxy_fix              | 0.1.2        |
| sawyer                    | 0.8.1        |
| octokit                   | 4.9.0        |
| unicode-display_width     | 1.4.0        |
| terminal-table            | 1.8.0        |
| danger                    | 5.6.1        |
| ox                        | 2.9.2        |
| danger-junit              | 0.7.4        |
| danger-plugin-api         | 1.0.0        |
| thor                      | 0.20.0       |
| danger-swiftlint          | 0.17.0       |
| danger-xcode_summary      | 0.5.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| dotenv                    | 2.5.0        |
| emoji_regex               | 0.1.1        |
| excon                     | 0.62.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.4        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.13.1       |
| os                        | 1.0.0        |
| signet                    | 0.9.2        |
| googleauth                | 0.6.6        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2018.0812  |
| mime-types                | 3.2.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.4.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.0        |
| tty-spinner               | 0.8.0        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| xml-simple                | 1.1.5        |
| fastlane-plugin-bugsnag   | 1.3.2        |
| trainer                   | 0.7.0        |
| fastlane-plugin-trainer   | 0.4.0        |
| mini_portile2             | 2.3.0        |
| nokogiri                  | 1.8.2        |
| phraseapp-ruby            | 1.6.0        |
| psych                     | 2.2.4        |
| xcpretty-json-formatter   | 0.1.0        |
</details>

*generated on:* **2018-09-21**
</details>

**Comment by sorianog** [comment]
Would be interested to grab a sample project to see if I can reproduce this on my machine (High Sierra).

**Comment by jinthagerman** [comment]
also looks like test reports aren't working either

**Comment by rockbruno** [comment]
@jinthagerman Can confirm, I have an unit test and an UI test module but only the unit tests are being reported when we run fastlane in Xcode 10

**Comment by jinthagerman** [comment]
So I looked into this further. It looks like `xcpretty` is actually at fault because the test output is formatted differently for parallel tests. https://github.com/supermarin/xcpretty/issues/295

I had some success using trainer instead of xcpretty's junit output, but xcodebuild still seems to fail in a way that makes scan fail even if its only a test failure (i.e `fail_build` is false).

**Comment by TomCarey** [comment]
@jinthagerman I get the same results as you. I've had to resort to running `xcodebuild` directly (as in not through scan) as I couldn't figure out how to get scan not to use `xcpretty`. I also had to ignore the exit status of the `xcodebuild` command, and rely on trainer instead.

**Comment by jinthagerman** [comment]
`output_types: '', output_style: 'raw'` should turn off xcpretty formatting and junit output and then I just wrapped scan in a `begin`/`ensure`.

**Comment by TomCarey** [comment]
awesome, thanks for that!

**Comment by FabienLydoire** [comment]
I've been using  https://github.com/KrauseFx/trainer and https://github.com/TitouanVanBelle/XCTestHTMLReport in replacement of the Scan JUnit output and the XCpretty report. Seems to be working with Xcode 10 and parallel testing...

**Comment by benasher44** [comment]
Okay so I narrowed this down some more. The issue is actually that I'm running into some kind of XCTest bug where a test failing always causes one of the sims to have an "early unexpected exit":

```
2018-10-26 13:18:29.224 xcodebuild[32927:407970] Error Domain=IDETestOperationsObserverErrorDomain Code=6 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted, NSUnderlyingError=0x7f9a6bbe7a20 {Error Domain=IDETestOperationsObserverErrorDomain Code=5 "Test runner exited before starting test execution." UserInfo={NSLocalizedDescription=Test runner exited before starting test execution., NSLocalizedRecoverySuggestion=If you believe this error represents a bug, please attach the result bundle at /Users/basher/Code/plangrid/DerivedData/Logs/Test/Test-PlanGrid_CI-2018.10.26_13-16-50--0700.xcresult}}}
2018-10-26 13:18:29.224 xcodebuild[32927:407970] Error Domain=IDETestOperationsObserverErrorDomain Code=6 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted, NSUnderlyingError=0x7f9a6be43860 {Error Domain=IDETestOperationsObserverErrorDomain Code=5 "Test runner exited before starting test execution." UserInfo={NSLocalizedDescription=Test runner exited before starting test execution., NSLocalizedRecoverySuggestion=If you believe this error represents a bug, please attach the result bundle at /Users/basher/Code/plangrid/DerivedData/Logs/Test/Test-PlanGrid_CI-2018.10.26_13-16-50--0700.xcresult}}}

Testing failed:
PlanGrid.app (33051) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. Underlying error: Test runner exited before starting test execution.)
PlanGrid.app (33066) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. Underlying error: Test runner exited before starting test execution.)
```

Because this prints `Testing failed`, that triggers  https://github.com/fastlane/fastlane/blob/master/scan/lib/scan/error_handler.rb#L25. Going to file an Apple bug.

**Comment by benasher44** [comment]
So there's an Xcode 10 bug here, but there's a `scan`/`run_tests` one too. Scan looks for the line that contains `Executed` in `Scan::ErrorHandler::handle_build_error`, but with Xcode 10 parallel testing, this line isn't printed, unless Xcode decides to run a particular test suite serially. Instead, fastlane needs to also check for `Failing tests:` in the "is this a build failure or test failure" scenario because Xcode will print out the tests that failed at the end of the run. Here's a patch you can put in your Fastfile to account for this:

```
fastlane_require 'scan'
module ::Scan
  class ErrorHandler
    class << self
      alias_method :orig_handle_build_error, :handle_build_error
      def handle_build_error(output)
        # Only call the original handle_build_error method if we can tell that there weren't failing tests
        orig_handle_build_error(output) if (output =~ /Failing tests:/).nil?
      end
    end
  end
end
```
However, this isn't enough, while this Xcode bug persists, but this is something that'll need to be updated in `handle_build_error` at some point.

**Comment by benasher44** [comment]
@KrauseFx as maintainer of trainer, maybe you have some insight into a better method of detecting "test runner failure" vs "build failure" vs "test ran and failed", possibly via inspecting plists? That might be more robust. Checking log output is now less reliable in the parallel testing world.

**Comment by benasher44** [comment]
After a response from Apple about our "bug," it turns out we were actually running into crashes during bootstrapping, which were caused by a crash in our application. The logs are just a bit tricky to find, but they are there in the xcresult bundle. I'm feeling more comfortable opening a PR to also include the log test for `Failing tests:`, which can appear after parallel testing fails due to failing tests.

**Comment by benasher44** [comment]
You’ll need to use trainer to get junit output from your tests when parallel testing is enabled. When parallel testing is enabled, Xcode doesn’t print out the standard test result output that xcpretty would be able to parse and turn into a junit file
